app/components/AdditionalData/index.tsx

```diff
@@ -0,0 +1,306 @@
+    {
+      key: 'import',
+      iconType: 'FA5Icon',
+      iconName: 'file-import',
+      text: 'label.import_data',
+    },
+    {
+      key: 'export',
+      iconType: 'FA5Icon',
+      iconName: 'file-export',
+      text: 'label.export_data',
+    },
+  ];
+
+  useEffect(() => {
+    const unsubscribe = navigation.addListener('focus', () => {
+      if (isInitialLoading) {
+        addFormsToState();
+        addMetadataInState();
+        setIsInitialLoading(false);
+      } else {
+        if (routeIndex === 0) {
+          addFormsToState();
+        } else if (routeIndex === 1) {
+          addMetadataInState();
+        }
+      }
+    });
+    return unsubscribe;
+  }, [navigation]);
+
+  useEffect(() => {
+    if (registrationType === OFF_SITE) {
+      let treeOptions = initialTreeTypeOptions;
+      const lastIndex = treeOptions.length - 1;
+      treeOptions[lastIndex] = { ...treeOptions[lastIndex], disabled: true };
+      setTreeTypeOptions(treeOptions);
+      if (treeType === SAMPLE) {
+        setSelectedTreeOption(treeOptions[0]);
+        setTreeType('all');
+      }
+    } else {
+      setTreeTypeOptions(initialTreeTypeOptions);
+    }
+    updateStateFormData(JSON.parse(JSON.stringify(forms)));
+  }, [treeType, registrationType]);
+
+  const updateStateFormData = (formsData: any) => {
+    formsData = filterFormByTreeAndRegistrationType(formsData, treeType, registrationType);
+    setFilteredForm(formsData);
+  };
+
+  const addFormsToState = () => {
+    setFormLoading(true);
+    getForms().then((formsData: any) => {
+      if (formsData) {
+        formsData = sortByField('order', formsData);
+        setForms(formsData);
+        updateStateFormData(formsData);
+      }
+      setFormLoading(false);
+    });
+  };
+
+  const addMetadataInState = () => {
+    setMetadataLoading(true);
+    getMetadata().then((data: any) => {
+      if (data) {
+        setMetadata(sortByField('order', data));
+      }
+      setMetadataLoading(false);
+    });
+  };
+
+  const updateMetadataOrder = async (updatedMetadata: any) => {
+    await updateMetadata(updatedMetadata).then((success: boolean) => {
+      if (success) {
+        addMetadataInState();
+      }
+    });
+  };
+
+  const addNewForm = async () => {
+    await addForm({ order: Array.isArray(filteredForm) ? filteredForm.length + 1 : 1 });
+    addFormsToState();
+  };
+
+  const renderScene = SceneMap({
+    form: () => (
+      <Form
+        {...{
+          filteredForm,
+          addFormsToState,
+          loading: formLoading,
+          registrationTypeOptions,
+          setRegistrationType,
+          treeTypeOptions,
+          setTreeType,
+          selectedTreeOption,
+          addNewForm,
+        }}
+      />
+    ),
+    metadata: () => (
+      <Metadata
+        {...{ metadata, updateMetadataOrder, addMetadataInState, loading: metadataLoading }}
+      />
+    ),
+  });
+
+  const handleImportExport = async (option: OptionsType) => {
```

Comment:
For Android we need to check for the right permission to be granted: https://github.com/Plant-for-the-Planet-org/treemapper/blob/8fe00c0183ca3f29e69257cfe86b5c005d8ba7d0/app/components/InventoryOverview/index.js#L281-L288

Reply:
This just needs to be check for the export and write permission, not import needing only read permission.